managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3189,15 +3247,32 @@ private CompletableFuture<Void> completeLedgerInfoForOffloaded(long ledgerId, UU
                                        }
                                    })
             .whenComplete((result, exception) -> {
-                    if (exception == null) {
+                if (injection != null) {
+                    lastOffloadCompleteFailed = true;
+                    refreshedIfOffloadCompleteFailed = false;
+                    injection.throwException(ledgerId);
+                }
+                if (exception == null) {
                         log.info("[{}] End Offload. ledger={}, uuid={}", name, ledgerId, uuid);
                     } else {
+                        lastOffloadCompleteFailed = true;
+                        refreshedIfOffloadCompleteFailed = false;
                         log.warn("[{}] Failed to complete offload of ledger {}, uuid {}",
                                  name, ledgerId, uuid, exception);
                     }
                 });
     }
 
+    private Injection injection;
```

Comment:
please do not add this mechanism on core classes.
we can use Mockito to interact with the internals and simulate problems